e2e/runner/zeta.go

```diff
@@ -31,6 +34,30 @@ func (r *E2ERunner) WaitForBlocks(n int64) {
 	err = retry.DoWithBackoff(call, boWithMaxRetries)
 	require.NoError(r, err, "failed to wait for %d blocks", n)
 }
+
+// WaitForTssGeneration waits for a specific number of TSS to be generated
+// The parameter n is the number of TSS to wait for
+func (r *E2ERunner) WaitForTssGeneration(n int64) {
+	call := func() error {
+		return retry.Retry(r.waitForTssGeneration(n))
+	}
+	bo := backoff.NewConstantBackOff(time.Second * 5)
+	boWithMaxRetries := backoff.WithMaxRetries(bo, 10)
+	err := retry.DoWithBackoff(call, boWithMaxRetries)
+	require.NoError(r, err, "failed to wait for %d tss generation", n)
+}
+
+func (r *E2ERunner) waitForTssGeneration(n int64) error {
```

Comment:
Name misleading, we don't wait in this function, we directly fetch and check list